pkg/middleware/requestid/requestid.go

```diff
@@ -45,15 +57,15 @@ func NewStreamingInterceptor() grpc.StreamServerInterceptor {
 
 func reportable() interceptors.CommonReportableFunc {
 	return func(ctx context.Context, c interceptors.CallMeta) (interceptors.Reporter, context.Context) {
-		id, _ := uuid.NewRandom()
-		requestID := id.String()
+		requestID := InitID(ctx)
 
 		ctx = metadata.AppendToOutgoingContext(ctx, requestIDCtxKey, requestID)
```

Comment:
I don't think we need this line, nor `requestIDCtxKey`. Can you remove them?

Reply:
- Regarding keeping `requestIDCtxKey`, Can we keep it and have different keys for different contexts?
Distinct key names may avoid potential conflicts or confusion between different contexts and their purposes. 

It will be aligned to what's done for naming keys in storeid middleware:
https://github.com/openfga/openfga/blob/abf56641636fddef9340c4462364de1505c30442/pkg/middleware/storeid/storeid.go#L17C1-L19C35 

- Regarding keeping the line, note that currently `func InitID()` is not setting any context key/value, all key/value setting is currently centralized inside `func reportable()`.
Some alternatives I see include
a) [recommended] keeping it as it is - solution is simple & more readable
b) refactor `func initCtxWithId()` to also append id to context metadata and return the updated ctx
c) mirror store.id middleware. - IMHO this option seems to introduce too much change that might not be worth it at this PR but could be evaluated for future refactor

Reply:
Any thoughts on the above proposals?

Reply:
This is how it works today on `main`:

1. On `requestid` we call `metadata.AppendToOutgoingContext(ctx, requestIDCtxKey, requestID)`
2. On `logging`, we call `if requestID, ok := requestid.FromContext(ctx); ok { fields = append(fields, zap.String(requestIDKey, requestID)) }`

Since you removed line 2, there is no reason to keep line 1 anymore, i.e. we don't need to keep the request id as a value in the context - that's what `grpc_ctxtags` is for.

Reply:
Thanks for clarifying that request-id should no longer be stored in ctx. Changes applied